### PR TITLE
Fully working and AWS-tested Lambda single-lookup function

### PR DESCRIPTION
**Pull request by russellwhitaker**

### CODE REVIEW

File: src/uap_clj_lambda/core.clj [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  (:require [uswitch.lambada.core :refer [deflambdafn]]
+            [clojure.data.json :as json :refer [read write]]
+            [clojure.java.io :as io :refer [reader writer]]
+            [uap-clj.core :refer [lookup-useragent]]))
+
+(defn handle-event
+  [event]
+  (lookup-useragent (get-in event ["useragent"])))
+
+(deflambdafn uap-clj-lambda.core.SimpleUseragentLookup
```
Comment (simshanith) [inline src/uap_clj_lambda/core.clj]: Is there a way to test lambda functions? It appears the spec only tests the `handle-event` function.

**Comment by simshanith** [comment]
LGTM :+1:

File: src/uap_clj_lambda/core.clj (hunk 1 above)
Comment (russellwhitaker) [inline src/uap_clj_lambda/core.clj]: Annoyingly, no, not yet. I spent hours looking in vain for *real* isolated unit test options, and they don't exist. Amazon's docs talk about functional testing using DynamoDB, but in that case, for this, might as well manually test. I'm not at all satisfied with this situation - I'm able to unit test the Hadoop wrapper version of this in https://github.com/russellwhitaker/uap-clj-hiveudf/tree/master/spec/uap_clj/udf/hive properly - so in the meantime, this is it.

**Comment by russellwhitaker** [comment]
Thanks @dhallman I'm a big believer in comprehensive README documentation.

**Comment by dhallman** [comment]
LGTM.

**Comment by dhallman** [comment]
@russellwhitaker - while this code LGTM and is a very useful example to gain experience with AWS Lambda, it also begs the question is this representative of the kinds of functionality we should be exposing through lambda? I certainly see huge value in Lambda, but I'm not clear on how it benefits this particular use-case, beyond direct invocation (i.e. via Hive UDF, for example). For example, I know Lambda exposure would allow Client Portal/Player to access UAP data, but those tools are on the generating end of that information, not the receiving end. So, I'm not clear on why they'd ever need to decode an individual user agent. Not being negative here, just trying to make sure we have a good answer to the "why" of this, not just the "how".

**Comment by russellwhitaker** [comment]
Good question @dhallman and I think @simshanith might have an opinion on this. Have you thought of use cases such as what a video player might use as a callback? Shane has. And it's not actually intended for direct invocation in production. For example, it just took me a minute to attach an API-key protected Amazon API Gateway POST endpoint to this, allowing:

→ curl -H "Content-Type: application/json" \
>      -H "x-api-key: ANAPIKEYIUSED" \
>      -X POST \
>      -d "{\"useragent\": \"Lenovo-A288t_TD\/S100 Linux\/2.6.35 Android\/2.3.5 Release\/02.29.2012 Browser\/AppleWebkit533.1 Mobile Safari\/533.1 FlyFlow\/1.4\"}" \
>      https://<private_instance>.execute-api.us-east-1.amazonaws.com/test/<my>-useragent-lookup

{"ua":"Lenovo-A288t_TD\/S100 Linux\/2.6.35 Android\/2.3.5 Release\/02.29.2012 Browser\/AppleWebkit533.1 Mobile Safari\/533.1 FlyFlow\/1.4","browser":{"family":"Baidu Explorer","major":"1","minor":"4","patch":""},"os":{"family":"Android","major":"2","minor":"3","patch":"5","patch_minor":""},"device":{"family":"Lenovo A288t_TD","brand":"Lenovo","model":"A288t_TD"}}

**Pull request merged**